### PR TITLE
solves failing benchmark analysis.rms

### LINKED ISSUE #3520: Benchmark for the RMS module fail
## Expected behavior ##

Benchmarks run without errors.

## Actual behavior ##

The benchmarks for `analysis.rms` fail with the following exceptions:

```
For parameters: 100, [1.0, 0.5], False, False
Traceback (most recent call last):
File "/home/benchy/mambaforge/lib/python3.9/site-packages/asv/benchmark.py", line 1184, in main_run_server
 main_run(run_args)
File "/home/benchy/mambaforge/lib/python3.9/site-packages/asv/benchmark.py", line 1058, in main_run
 result = benchmark.do_run()
File "/home/benchy/mambaforge/lib/python3.9/site-packages/asv/benchmark.py", line 537, in do_run
 return self.run(*self._current_params)
File "/home/benchy/mambaforge/lib/python3.9/site-packages/asv/benchmark.py", line 627, in run
 samples, number = self.benchmark_timing(timer, min_repeat, max_repeat,
File "/home/benchy/mambaforge/lib/python3.9/site-packages/asv/benchmark.py", line 694, in benchmark_timing
 timing = timer.timeit(number)
File "/home/benchy/mambaforge/lib/python3.9/timeit.py", line 177, in timeit
 timing = self.inner(it, self.timer)
File "<timeit-src>", line 6, in inner
File "/home/benchy/mambaforge/lib/python3.9/site-packages/asv/benchmark.py", line 599, in <lambda>
 func = lambda: self.func(*param)
File "/home/benchy/mdanalysis/benchmarks/benchmarks/analysis/rms.py", line 46, in time_rmsd
 rms.rmsd(a=self.A,
File "/home/benchy/mambaforge/lib/python3.9/site-packages/MDAnalysis/analysis/rms.py", line 260, in rmsd
 raise ValueError('weights must have same length as a and b')
ValueError: weights must have same length as a and b

For parameters: 100, [1.0, 0.5], False, True
Traceback (most recent call last):
File "/home/benchy/mambaforge/lib/python3.9/site-packages/asv/benchmark.py", line 1184, in main_run_server
 main_run(run_args)
File "/home/benchy/mambaforge/lib/python3.9/site-packages/asv/benchmark.py", line 1058, in main_run
 result = benchmark.do_run()
File "/home/benchy/mambaforge/lib/python3.9/site-packages/asv/benchmark.py", line 537, in do_run
 return self.run(*self._current_params)
File "/home/benchy/mambaforge/lib/python3.9/site-packages/asv/benchmark.py", line 627, in run
 samples, number = self.benchmark_timing(timer, min_repeat, max_repeat,
File "/home/benchy/mambaforge/lib/python3.9/site-packages/asv/benchmark.py", line 694, in benchmark_timing
 timing = timer.timeit(number)
File "/home/benchy/mambaforge/lib/python3.9/timeit.py", line 177, in timeit
 timing = self.inner(it, self.timer)
File "<timeit-src>", line 6, in inner
File "/home/benchy/mambaforge/lib/python3.9/site-packages/asv/benchmark.py", line 599, in <lambda>
 func = lambda: self.func(*param)
File "/home/benchy/mdanalysis/benchmarks/benchmarks/analysis/rms.py", line 46, in time_rmsd
 rms.rmsd(a=self.A,
File "/home/benchy/mambaforge/lib/python3.9/site-packages/MDAnalysis/analysis/rms.py", line 255, in rmsd
 a = a - np.average(a, axis=0, weights=weights)
File "<__array_function__ internals>", line 180, in average
File "/home/benchy/mambaforge/lib/python3.9/site-packages/numpy/lib/function_base.py", line 515, in average
 raise ValueError(
ValueError: Length of weights not compatible with specified axis.
```

### PR DESCRIPTION
**Pull request by ooprathamm**
Fixes #3520

PR Checklist
------------
 - [ ] Tests?
 - [ ] Docs?
 - [ ] CHANGELOG updated?
 - [ ] Issue raised/referenced?

### CODE REVIEW

File: benchmarks/benchmarks/analysis/rms.py [hunk 1]
```diff
@@ -15,7 +16,7 @@ class SimpleRmsBench(object):
     """
 
     params = ([100, 500, 2000],
-              [None, [1.0, 0.5]],
+              [None, [True]],
```
Comment (RMeli) [inline benchmarks/benchmarks/analysis/rms.py:19]:
```suggestion
              [True, False],
```

File: benchmarks/benchmarks/analysis/rms.py [hunk 2]
```diff
@@ -33,10 +34,13 @@ def setup(self, num_atoms, weights, center, superposition):
             self.A = self.u.atoms.positions.copy()[:num_atoms]
             self.u.trajectory[-1]
             self.B = self.u.atoms.positions.copy()[:num_atoms]
+            self.weights = [self.A.masses() / np.sum(self.A.masses()),self.B.masses() / np.sum(self.B.masses())] if 'weights' in self.params[1] else None
```
Comment (RMeli) [inline benchmarks/benchmarks/analysis/rms.py:37]:
```suggestion
            self.weights = [self.A.masses() / np.sum(self.A.masses()),self.B.masses() / np.sum(self.B.masses())] if weights else None
```
I feel the `weights` parameter should now be renamed to `use_weights`, for clarity.

File: benchmarks/benchmarks/analysis/rms.py [hunk 3]
```diff
@@ -33,10 +34,13 @@ def setup(self, num_atoms, weights, center, superposition):
...
+
         except:
             self.A = self.u.atoms.positions.copy()[:num_atoms]
             self.u.trajectory[-1]
             self.B = self.u.atoms.positions.copy()[:num_atoms]
+            self.weights = [self.A.masses() / np.sum(self.A.masses()),self.B.masses() / np.sum(self.B.masses())] if 'weights' in self.params[1] else None
```
Comment (RMeli) [inline benchmarks/benchmarks/analysis/rms.py:43]: Maybe I'm missing something obvious here, but I think the whole `try`/`except` block is superfluous now? Both blocks have the same code, after #3668.

File: benchmarks/benchmarks/analysis/rms.py [hunk 4]
```diff
@@ -45,7 +49,7 @@ def time_rmsd(self, num_atoms, weights, center, superposition):
         """
         rms.rmsd(a=self.A,
                  b=self.B,
-                 weights=weights,
+                 weights=self.weights,
```
Comment (RMeli) [inline benchmarks/benchmarks/analysis/rms.py:52]: The `weights` parameter in `time_rmsd` is now ignored and should be removed.

**Review by RMeli — CHANGES_REQUESTED** [review]

**Comment by orbeckst** [comment]
I haven't seen any progress on the PR so I am closing it. It's a bit sad to close a PR that's pretty far along and has detailed comments on how to improve it. However, if anyone wants to pick up the work, they are more than welcome and we can reopen the PR.

**Pull request closed without merging**